net/mwan3/files/usr/sbin/mwan3track

```diff
@@ -136,14 +143,23 @@ main() {
 			if [ $host_up_count -lt $reliability ]; then
 				case "$track_method" in
 					ping)
+						# Ensure that IPv6 address is reset
+						unset ADDR
 						if echo $track_ip | grep -q ':'; then
+							if [ "$FIX_KERNEL_ACTUAL_SK_BUG" = true ]; then
```

Comment:
To fix the use of `sh` natural booleans:
```suggestion
							if "$FIX_KERNEL_ACTUAL_SK_BUG"; then
```

But more importantly, isn't this logic inverted?  Doesn't `$FIX_KERNEL_ACTUAL_SK_BUG` being true mean that we no longer need to use the below `$ADDR` logic on lines 150-155?  And therefore this should actually be:
```suggestion
							if ! "$FIX_KERNEL_ACTUAL_SK_BUG"; then
```

Reply:
I would say that fits. The code will only be processed if we are to fix the kernel bug. And this is only true if the version of the openwrt kernel is less than 4.14.209 in the openwrt-19.07 branch.

What's wrong with ADDR? I don't understand your statement.

Reply:
What I am saying is that we **don't** want to use the `$ADDR` method of `ping -I` if the kernel is `> 4.14.209`, right?  Isn't the whole point that newer kernels have removed the need for this?

The logic as is is using `ping -I $ADDR` when the kernel is `> 4.14.209`.  Is that what we want?